crates/resolver/src/dns_sd.rs

```diff
@@ -59,7 +58,9 @@ impl DnsSdHandle for AsyncResolver {
 }
 
 /// A DNS Service Discovery future of Services discovered through the list operation
-pub struct ListServicesFuture(Pin<Box<BackgroundLookup<ReverseLookupFuture>>>);
+pub struct ListServicesFuture(
+    Pin<Box<dyn Future<Output = Result<ReverseLookup, ResolveError>> + Send>>,
```

Comment:
Probably want to ensure that these futures are 'static iirc trait objects are a bit weird and all the runtimes require `'static`, I could be wrong though.

Reply:
Interesting.

Reply:
Ok, I've added 'static to those types so that we don't end up with odd issues down the line.